common/http/platform/beast/http.cpp

```diff
@@ -235,28 +235,34 @@ Client::Client(
 	const ClientConfig &client, events::EventLoop &event_loop, const string &logger_name) :
 	event_loop_ {event_loop},
 	logger_name_ {logger_name},
+	http_proxy_ {client.http_proxy},
+	https_proxy_ {client.https_proxy},
+	no_proxy_ {client.no_proxy},
 	cancelled_ {make_shared<bool>(true)},
 	disable_keep_alive_ {client.disable_keep_alive},
 	resolver_(GetAsioIoContext(event_loop)),
 	body_buffer_(HTTP_BEAST_BUFFER_SIZE) {
-	ssl_ctx_.set_verify_mode(client.skip_verify ? ssl::verify_none : ssl::verify_peer);
-
-	if (client.client_cert_path != "" and client.client_cert_key_path != "") {
-		ssl_ctx_.set_options(boost::asio::ssl::context::default_workarounds);
-		ssl_ctx_.use_certificate_file(client.client_cert_path, boost::asio::ssl::context_base::pem);
-		ssl_ctx_.use_private_key_file(
-			client.client_cert_key_path, boost::asio::ssl::context_base::pem);
-	}
+	for (auto i = 0; i < MENDER_BOOST_BEAST_SSL_CTX_COUNT; i++) {
+		ssl_ctx_[i].set_verify_mode(client.skip_verify ? ssl::verify_none : ssl::verify_peer);
+
+		if (client.client_cert_path != "" and client.client_cert_key_path != "") {
+			ssl_ctx_[i].set_options(boost::asio::ssl::context::default_workarounds);
+			ssl_ctx_[i].use_certificate_file(
```

Comment:
Note that this will throw an exception in the case of a missing file. 

Same below

Reply:
Pre-existing, but indeed, good catch! I'll fix both of them.